### PR TITLE
A_Quake(Ex): Intensity int-to-double

### PR DESCRIPTION
**Pull request by MajorCooke**
Converted A_Quake(Ex) intensity parameters from ints to doubles.

### CODE REVIEW

File: src/playsim/mapthinkers/a_quake.cpp [hunk 1]
```diff
@@ -408,17 +408,17 @@ int DEarthquake::StaticGetQuakeIntensities(double ticFrac, AActor *victim, FQuak
...
 	double rollIntensity, double rollWave)
 {
 	AActor *center;
 	bool res = false;
 
-	if (intensityX)		intensityX = clamp(intensityX, 1, 9);
-	if (intensityY)		intensityY = clamp(intensityY, 1, 9);
-	if (intensityZ)		intensityZ = clamp(intensityZ, 1, 9);
+	intensityX = clamp<double>(intensityX, 0.0, 9.0);
+	intensityY = clamp<double>(intensityY, 0.0, 9.0);
```
Comment (CandiceJoy) [inline src/playsim/mapthinkers/a_quake.cpp:420]: Is there a particular reason this is 0.0 to 9.0 instead of 1-9?

**Review by CandiceJoy — COMMENTED** [review]
The parameters come in from ACS as floats and get implicitly converted to doubles.  While this doesn't introduce any errors, there's really no need for this to be double, I think.  In my opinion, it'd be better to use float throughout, both for efficiency and for consistency, plus it's been int for so long, I don't think we really need the precision of a double :P

**Comment by mjr4077au** [comment]
> The parameters come in from ACS as floats and get implicitly converted to doubles. 

The general rule of thumb is double inside the playsim and float outside the playsim, at least with GZDoom and Raze.

**Comment by CandiceJoy** [comment]
> > The parameters come in from ACS as floats and get implicitly converted to doubles.
> 
> The general rule of thumb is double inside the playsim and float outside the playsim, at least with GZDoom and Raze.

Does that include situations where a variable gets upcast from float to double during the transition?

**Comment by mjr4077au** [comment]
I believe so. For instance in Raze, the player's forward/side velocity is a float but when it gets rotated by their angle, it becomes a double.

**Comment by CandiceJoy** [comment]
Fair enough.  Seems a bit odd to me, but if that's the way it is, that's the way it is :)

File: src/playsim/mapthinkers/a_quake.cpp (hunk 1 above)
Comment (inkoalawetrust) [reply src/playsim/mapthinkers/a_quake.cpp:420]: It allows for setting sub 1.0 intensities, like a really small earthquake with an intensity of 0.5.

**Pull request merged**